### PR TITLE
feat(@formatjs/intl-displaynames): Intl DisplayNames API V2 (Stage 3) implementation

### PR DESCRIPTION
**Pull request by andreaforni**
This PR implements the [Intl.DisplayNames v2 Proposal (Stage 3)](https://tc39.es/proposal-intl-displaynames-v2/).

The PR works mainly in the `packages/intl-displaynames` package. In details:
- It implements the new `IsValidDateTimeFieldCode` abstract operation and provides unit test for it.
- It updates the `CanonicalCodeForDisplayNames` abstract operation to manage the two new types `calendar` and `dateTimeField`. It provides unit tests for it.
- It updates the `DisplayNamesOptions` implementation to manage:
   - The `calendar` and  `dateTimeField` fields.
   - The new `languageDisplay` option. This option controls how to display the name of a language. It has two values: `standard` and `dialect` (default).
- It adds new tests to check the new features introduced in the previous point. Please check the `packages/intl-displaynames/tests/index.test.ts` file.
- It updates the `extract-displaynames.ts` script to generate the new localization files. Now, the scripts also extract data related to `calendar`, `dateTimeField`, `standard` and `dialect` language.
  - The changes in the script above,  required regenerating the `test262-main.ts`, `tests/locale-data/en.json` and `tests/locale-data/zh.json` fiels.

The PR also update the tests and code related to the display name in the `packages/intl` package.

cc @micheleriva

### CODE REVIEW

**Comment by simoneb** [comment]
@longlho apologies for the direct ping, but the buildbuddy check seems to be failing for reasons unrelated to the changes in this branch. It failed when fetching a git submodule apparently. Any chance you can retrigger it?

File: packages/intl-displaynames/tests/index.test.ts [hunk 1]
```diff
@@ -8,45 +8,142 @@ import * as zh from './locale-data/zh.json'
...
+            'xx-CN'
+          )
+        ).toBe(undefined)
+      })
+    })
   })
 
   it('accepts case-insensitive currency code', () => {
     expect(
-      new DisplayNames('en-US', {type: 'currency', style: 'long'}).of('cNy')
```
Comment (longlho) [inline packages/intl-displaynames/tests/index.test.ts]: why this change?

Comment (andreaforni) [reply packages/intl-displaynames/tests/index.test.ts]: Because the value of the `locale` field inside the `./locale-data/en.json` file is `en`, not `en-US`; so English localization data are bound to the `en` locale. For this reason I thought to correct the value in this test

Comment (longlho) [reply packages/intl-displaynames/tests/index.test.ts]: `en` & `en-US` canonicalize to the same thing. If you keep it as `en-US` does it crap out?

Comment (andreaforni) [reply packages/intl-displaynames/tests/index.test.ts]: It works with `en-US` too. Do you want me to revert it to `en-US` and update all the new tests to use `en-US`?

Comment (longlho) [reply packages/intl-displaynames/tests/index.test.ts]: I think just revert this one for sanity check. There's some locale matching mechanism underneath the hood and I wanna make sure it doesn't change that

Comment (andreaforni) [reply packages/intl-displaynames/tests/index.test.ts]: Done

**Comment by longlho** [comment]
Thanks a lot for your contributions!
